src/backend/context.rs

```diff
@@ -193,26 +215,28 @@ impl Context {
     }
 
     pub fn max_channel_count(&self) -> Result<u32> {
-        unsafe {
-            self.mainloop.lock();
-            while self.default_sink_info.is_null() {
-                self.mainloop.wait();
-            }
-            self.mainloop.unlock();
+        self.mainloop.lock();
+        while self.default_sink_info.is_none() {
```

Comment:
I ended up implementing that in another branch (when preparing the old version to land in gecko)